### PR TITLE
Add a version specifier parameter

### LINKED ISSUE #8186: Beat allow to pass build time version qualifier
Allow passing a build time qualifier so that a build artifact can be promoted to GA without requiring a rebuild. Our build process accepts the `SNAPSHOT` environment variable to add the snapshot suffix, and we need to add a `VERSION_QUALIFIER` options that will allow the release manager to specify the qualifier to be used.

The generated package will use the specified qualifier and the beats will also report their version using the qualifier used at built time.

### PR DESCRIPTION
**Pull request by tsg**
The VERSION_QUALIFIER env variable adds a custom text after the version.
Example run:

     VERSION_QUALIFIER=rc1 mage -v package

Closes #8186.

### CODE REVIEW

File: dev-tools/mage/settings.go [hunk 1]
```diff
@@ -145,18 +147,20 @@ func varMap(args ...map[string]interface{}) map[string]interface{} {
...
+BinaryExt        = {{.BinaryExt}}
+BeatName         = {{.BeatName}}
+BeatServiceName  = {{.BeatServiceName}}
+BeatIndexPrefix  = {{.BeatIndexPrefix}}
+BeatDescription  = {{.BeatDescription}}
+BeatVendor       = {{.BeatVendor}}
+BeatLicense      = {{.BeatLicense}}
+BeatURL          = {{.BeatURL}}
+VersionQualifier = {{.VersionQualifier}}
+Snapshot         = {{.Snapshot}}
```
Comment (ruflin) [inline dev-tools/mage/settings.go]: Was that a bug?

File: filebeat/magefile.go [hunk 2]
```diff
@@ -82,8 +82,8 @@ func Package() {
 	mage.UseElasticBeatPackaging()
 	customizePackaging()
 
-	mg.Deps(Update, prepareModulePackaging)
-	mg.Deps(CrossBuild, CrossBuildXPack, CrossBuildGoDaemon)
+	//mg.Deps(Update, prepareModulePackaging)
```
Comment (ruflin) [inline filebeat/magefile.go]: Commented out on purpose?

Comment (tsg) [reply filebeat/magefile.go]: Good catch, no, that was for faster debugging.

File: dev-tools/mage/settings.go (hunk 1 above)
Comment (tsg) [reply dev-tools/mage/settings.go]: I think not really, but I think it makes sense to add it for completion.

**Comment by tsg** [comment]
@andrewkroh Compared to the version that we worked out together, I did two changes:

* renamed VersionSpecifier to VersionQualifier to mirror the naming from other projects
* I made VersionQualifier not need a dash (i.e. `VERSION_QUALIFIER='rc1'` instead of `VERSION_QUALIFIER='-rc1'`.

**Comment by tsg** [comment]
@jasontedor Ah, I must have misunderstood the requirements. I thought we're going to change the internal version to `7.0.0` (I didn't do it in this PR so that we don't break the build early) but only name the external package name to 7.0.0-rc1, 7.0.0-rc2, etc. So if, say, we consider 7.0.0-rc2 successful, we can just rename the artifacts without re-building.

I understand now that actually a rebuild is wanted to go from rc2 to ga, just that we will use the same commit hash for building. Thanks for clarifying, I'll update the PR to reflect that.

**Comment by andrewkroh** [comment]
I think we want to inject the version into to the build similar to Gil's branch or how we set the build time and commit ID values.

I don't like how we set the version in code (e.g. `const defaultBeatVersion = "7.0.0-alpha1"`). I'd much rather see the branch version contained in some kind of properties file and always be injected into the build. This would simplify the build logic IMO.

`Read unqualified version from file -> Append qualifier -> go build -X "version=7.0.0-rc1"`

**Comment by graphaelli** [comment]
@tsg per our discussion, I've opened #8310 for this.

**Comment by tsg** [comment]
Closing in favour of #8310. Thanks @graphaelli!

**Pull request closed without merging**